### ISSUE TITLE
Feature request: support Jest

### ISSUE THREAD

**Issue by trevordmiller** [algolia/expect-jsx#23]
The latest versions of Jest have improved a lot (no more auto-mocking, added snapshot testing and only testing changed files since last commit etc) and I see more developers using Jest in React projects; I'm converting a project from Mocha + Karma to Jest right now and everything just worked except for our use of `expect-jsx` since we are now using Jest's global `expect` instead of @mjackson's `expect` npm package. It looks like Jest extend's Jasmine's global `expect`: https://github.com/facebook/jest/blob/master/packages/jest-jasmine2/src/extendJasmineExpect.js. I don't know the technical implementation of `expect-jsx` but would it be easy/possible to support Jest's `expect`? I would take a whack at it myself but have a bit too much on my plate ATM.

p.s. Thank you for this great package; I've been using it for the last six months and enjoyed it. I even made an egghead lesson on it https://egghead.io/lessons/react-testing-jsx-error-diffs

cc: @cpojer

**Comment by vvo** [comment]
Nice! It should be feasible to support Jest's expect with expect-jsx but I am wondering, I build expect-jsx to assert JSX output.

While now, I am not anymore using it, I only use snapshot testing which is what I need in the end. Do you have needs for both expect-jsx and snapshot testing?

**Comment by trevordmiller** [comment]
@vvo Good point. I think we may just switch all our tests over to snapshot testing :)

**Comment by vvo** [comment]
Allright, let's close this issue then :)

**Comment by cpojer** [comment]
Has anyone tried to use Jest's `toEqual` with two React elements? It might work just fine! If not, I'm happy to accept any PRs that add functionality into Jest.

We do also pretty-print React elements by default, so that should make everything a bunch nicer.

**Comment by trevordmiller** [comment]
@cpojer I haven't tried that. However, I would definitely use snapshots for an a === a type now (so I don't really see value in a `toEqual` with two React elements as a snapshot is perfect). Where I've been using `expect-jsx` was with tests like this:
![image](https://cloud.githubusercontent.com/assets/5497885/18658392/b567e90a-7ebf-11e6-9062-9b4743116746.png)
As shown in my [React Testing Cookbook egghead course](https://egghead.io/courses/react-testing-cookbook)
I would assume I could either replace the `.toIncludeJSX` with a `String.includes` (without the pretty diff), or better yet just refactor to use snapshot tests; I like snapshot tests for all UI and more granular logic tests for non-UI.
